packages/manager/modules/hub/src/components/carousel/carousel.controller.js

```diff
@@ -1,8 +1,27 @@
+import { map, filter } from 'lodash-es';
+
 export default class {
   /* @ngInject */
-  constructor(atInternet) {
+  constructor($q, $http, $translate, atInternet) {
     this.atInternet = atInternet;
     this.index = 0;
+    this.$http = $http;
+    this.$q = $q;
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    if (!this.items || this.items.length === 0) {
```

Comment:
why check the items length ? if there is no notification why refetch them ?

Reply:
The idea was to leave the items binding if we want to call the component with items other than notifications( or if the caller already has notifications items). If the caller doesn't specify anything, we fetch notifications by default

Maybe we could change the condition or document it, after reading your comment, now i'm thinking that the caller could want to pass an empty items object, maybe bind a boolean instead? like "fetchNotifications" in the component to replace the condition?

Reply:
@zakariasahmane it depends on the usage but maybe we could just check `!this.items` and create a Breaking change on the component